pkg/util/nativeutils/republisher/republisher.go

```diff
@@ -5,9 +5,10 @@ import (
 
 	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/topics"
 	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus"
+	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/prerror"
 )
 
-type Validator func(bytes.Buffer) error
+type Validator func(bytes.Buffer) *prerror.PrError
```

Comment:
I somewhat dislike setting priorities during `error` creation. I get the impression that it is a practice that pollutes the interface by forcing implementers to handle information they should have no say about. Whether an error is high priority or not, is none of the Validator business, and it should be a responsibility left to the components that handle the validation errors. I would suggest to include some common validation errors (something like "DuplicatePayloadError", "EncodingError" and "InvalidError") and type-check/handle them at the caller level

Reply:
I see what you mean. I'll adjust it.